devices/Bq2579x/Bq2579x.cs

```diff
@@ -0,0 +1,939 @@
+        [Property]
+        public ThresholdFastCharge FastChargeTransitionVoltage { get => GetThresholdFastCharge(); set => SetThresholdFastCharge(value); }
+
+        /// <summary>
+        /// Gets or sets precharge current limit (in steps of 40mA).
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">If value is out of range (40mA-2000mA).</exception>
+        /// <remarks>Equivalent range: 40mA-2000mA.</remarks>
+        [Property]
+        public ElectricCurrent PrechargeCurrentLimit { get => GetPrechargeCurrentLimit(); set => SetPrechargeCurrentLimit(value); }
+
+        /// <summary>
+        /// Gets VBUS.
+        /// </summary>
+        /// <remarks>Range: 0mV-30000mV.</remarks>
+        public ElectricPotentialDc Vbus => GetAdcVoltage(Register.REG35_VBUS_ADC);
+
+        /// <summary>
+        /// Gets VAC1.
+        /// </summary>
+        /// <remarks>Range: 0mV-30000mV.</remarks>
+        public ElectricPotentialDc Vac1 => GetAdcVoltage(Register.REG37_VAC1_ADC);
+
+        /// <summary>
+        /// Gets VAC2.
+        /// </summary>
+        /// <remarks>Range: 0mV-30000mV.</remarks>
+        public ElectricPotentialDc Vac2 => GetAdcVoltage(Register.REG39_VAC2_ADC);
+
+        /// <summary>
+        /// Gets VBAT.
+        /// </summary>
+        /// <remarks>Range: 0mV-20000mV.</remarks>
+        public ElectricPotentialDc Vbat => GetAdcVoltage(Register.REG3B_VBAT_ADC);
+
+        /// <summary>
+        /// Gets VSYS.
+        /// </summary>
+        /// <remarks>Range: 0mV-24000mV.</remarks>
+        public ElectricPotentialDc Vsys => GetAdcVoltage(Register.REG3D_VSYS_ADC);
+
+        /// <summary>
+        /// Gets die temperature.
+        /// </summary>
+        /// <remarks>Range: -40�C -150�C.</remarks>
+        public Temperature DieTemperature => GetDieTemperature();
+
+        /// <summary>
+        /// Gets or sets the watchdog timer setting (in milliseconds).
+        /// </summary>
+        public WatchdogSetting WatchdogTimerSetting { get => GetWatchdogTimerSetting(); set => SetWatchdogTimerSetting(value); }
+
+        /// <summary>
+        /// Gets or sets Thermal regulation threshold.
+        /// </summary>
+        [Property]
+        public ThermalRegulationThreshold ThermalRegulationThreshold { get => GetThermalRegulationThreshold(); set => SetThermalRegulationThreshold(value); }
+
+        /// <summary>
+        /// Gets or sets Thermal shutdown threshold.
+        /// </summary>
+        [Property]
+        public ThermalShutdownThreshold ThermalShutdownThreshold { get => GetThermalShutdownThreshold(); set => SetThermalShutdownThreshold(value); }
+
+        /// <summary>
+        /// Gets Power Good Status.
+        /// </summary>
+        /// <value><see langword="true"/> if power is good, otherwise <see langword="false"/>.</value>
+        [Property]
+        public bool PowerGood => GetChargerStatus0().HasFlag(ChargerStatus0.PowerGood);
```

Comment:
Maybe `IsPowerGood`would be a better name?

Reply:
OK